### DISCUSSION TITLE
Uncertainties about Roles and Subroles

### DISCUSSION THREAD

**Discussion by JanEmde** [OP-TED/eForms-SDK#844] [Q&A] [answered]
The documentation about roles and subroles (https://docs.ted.europa.eu/eforms/latest/schema/parties.html) states that "Awarding CPB" and "Acquiring CPB" are independent authority roles. However, these "roles" can currently only be recorded as additional information to the "buyer" role. As I understand it, these roles should therefore be subroles of the role "buyer".

The documentation also states "subroles are only accessible to organisations which already have a given role assigned". However, this rule is not applied. There are countless notices in which organisations only have a "subrole". In general, sub-roles are not labelled as such in the Notice View. Instead, all roles (and subroles) are simply listed one below the other. 

A clarification regarding the (future) handling of roles and subroles would be very helpful.

**Comment by YvesJo** [comment, marked as answer]
Hi,
Possible sub-roles for a given role are listed in the first table of the following section: https://docs.ted.europa.eu/eforms/latest/schema/parties.html#_overview_of_roles_sub_roles_and_functions . Also any Organisation has to have at least one associated role.

**Reply by JanEmde** [reply]
Thanks for the answer. My questions are:

1. The roles "Awarding CPB" and "Acquiring CPB" can only be recorded as additional information to the "buyer" role? Shouldn't these roles be subroles?

2. The documentation also states "subroles are only accessible to organisations which already have a given role assigned". When will this rule be applied?

3. Will the TED notice view present subroles as such in the future?

**Reply by YvesJo** [reply]
Hi,
The distinction between the concepts of Roles and Sub-roles is mainly driven by their interdependency. The interdependency is controlled for each sub-role using Stage 5 (co-constraints) rules preventing incompatible assignments to a given Organisation.
For any specific need like adapting the view to distinguish Roles from Sub-roles, I invite you to get in touch with the Helpdesk instead: ted@publications.europa.eu
